workloadinterface/workloadmethods.go

```diff
@@ -691,3 +699,80 @@ func (w *Workload) GetPodStatus() (*corev1.PodStatus, error) {
 	}
 	return &status, nil
 }
+
+// GetVolumeMountsPaths returns all volume mounts paths of the workload
+// if readOnly is nil, returns all volume mounts paths
+// if readOnly is true, returns only readOnly volume mounts paths
+// if readOnly is false, returns only readWrite volume mounts paths
+// if mustHaveHostVolume is true, returns only volume mounts paths that have host volume
+func (w *Workload) GetVolumeMountsPaths(readOnly *bool, mustHaveHostVolume bool) ([]VolumeMount, error) {
+	podSpec, err := w.GetPodSpec()
+	if err != nil {
+		return nil, err
+	}
+
+	var hostVolumeNames []string
+
+	if mustHaveHostVolume {
+		hostVolumes, err := w.GetHostVolumes()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, hostVolume := range hostVolumes {
+			hostVolumeNames = append(hostVolumeNames, hostVolume.Name)
+		}
+	}
+
+	var volumeMounts []VolumeMount
+	for i, container := range podSpec.Containers {
+		if len(container.VolumeMounts) > 0 {
+			for j, volumeMount := range container.VolumeMounts {
+				if readOnly == nil || volumeMount.ReadOnly == *readOnly {
+					if !mustHaveHostVolume || (mustHaveHostVolume && slices.Contains(hostVolumeNames, volumeMount.Name)) {
+						volumeMounts = append(volumeMounts, VolumeMount{
+							ContainerIdx:    i,
+							VolumeMountIdx:  j,
+							VolumeMountPath: volumeMount.MountPath,
+						})
+					}
+				}
+			}
+		}
+
+	}
+
+	return volumeMounts, nil
+
+}
+
+// GetHostVolumes returns all host volumes of the workload
+func (w *Workload) GetHostVolumes() ([]v1.Volume, error) {
```

Comment:
Why do you have a specific function in this interface? I would expect a function `GetVolumes` and the caller of the function will filter the hostVolume or any other interested volumes.

Reply:
@dwertent host volume is quite fundamental so it make sense to me it can be on interface level. moreover, it also make sense to filter within loop rather than loop again the results for filtering.

How about GetVolumes(withHostPath *bool) to cover both cases?

Reply:
@kooomix 
The argument you are making here is:
1. HostPath is fundamental so we want a dedicated function for that in the interface level
2. We don't want to loop twice.

Here are my thoughts:
1. What other use case do you have where you need only the hostPath volume? What if you need the emptyDir volume or any other types of volumes? If hostPath is so fundamental, Kubernetes will have a kind for volumes and hostPathVolumes. But they don't.
2. How many volumes can be there that you do not want to loop twice over the list? I do not think it should be a concern in this case.

Adding a boolean for such an indication is also not a good solution because it is too specific.
You can add an argument for `volumeTypes` and return the types that are requested. If the list is empty you will return all types. e.g. `GetVolumes(volumeType []v1.VolumeType)` (I made up `v1.VolumeType`).
This solution could work although I don't like it because we are adding too much logic in a function that should simply list the volumes, but I will accept it if you decide to go with it.

Reply:
@dwertent ok, removed the function.

please note that I replaced an existing GetVolumes function with a new one, as the older one was dependant on a deprecated function.